pkgs/build-support/record-playback/record-command

```diff
@@ -0,0 +1,24 @@
+#! /bin/sh
+
+if [ "$1" = "--playback" ]; then
+  shift
+  PLAYBACK=1
+fi
+
+cmd="$1"; shift
+shortargs="$(printf %s "$*" | tr -d '[:space:]')"
```

Comment:
Command lines can potentially be larger than file names, `mkdir -p "$dir"` might fail to record the command line.